src/Entities/Entity.h

```diff
@@ -662,8 +662,8 @@ class cEntity
 
 	cWorld * m_World;
 
-	/** If not nullptr, a world change is scheduled and a task is queued in the current world. */
-	cAtomicUniquePtr<sWorldChangeInfo> m_WorldChangeInfo;
+	/** If field m_NewWorld not nullptr, a world change is scheduled and a task is queued in the current world. */
+	sWorldChangeInfo m_WorldChangeInfo;
```

Comment:
>I've made sWorldChangeInfo a plain member, the rationale being that entites should not be manipulated outside of the tick thread

That assumption is false. Read the comment I linked above.

Reply:
If it really enables plugins to call into an entity asynchronous to world tick then MoveToWorld is the least of our problems.

Could you explain where the chunkmap's lock comes into play here?

Reply:
notice me peterbell senpai

Reply:
If you made sure the chunk map CS was held during player chat handlers then this would be okay.

Reply:
Player chat is processed in the context of the world tick thread, so this is fine. Did you find any other cases during your work on world travel?

Reply:
I can't remember any changes touching that code so I doubt it's been fixed.